third_party/move/move-vm/types/src/value_serde.rs

```diff
@@ -62,11 +62,42 @@ impl DelayedFieldsExtension<'_> {
     }
 }
 
+/// Contains information on how to resolve function values.
+#[derive(Clone)]
+pub(crate) struct FunctionValueExtensionWithContext<'a> {
+    extension: &'a dyn FunctionValueExtension,
+    /// Marker to indicate that function value serialization failed. Used to ensure we propagate
+    /// error status code correctly, as otherwise any serialization failure is treated as an
+    /// invariant violation.
+    function_value_serialization_error: RefCell<Option<PartialVMError>>,
```

Comment:
So until now we basically considered serialization can never fail with a user level error, now with the check for delayed fields, it can. I wonder whether this has (unwanted) consequences and restricting us somehow (for example when implementing "fail fast"), but right now, it seems not viable to go any other way, apart of traversing everything ahead of serialization. So fine we go ahead with this.

Reply:
Yes, we already have some limits on number of delayed fields in a resource, in case of too many, an error is returned. Currently it is still fail fast: bcs will fail, but the marker is set to the right error code so that we can use it (bcs failure is mapped to option).